cmd/juno/dbsize.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/db/pebble"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/spf13/cobra"
+)
+
+func DBSize() *cobra.Command {
+	dbSizeCmd := &cobra.Command{
+		Use:   "db-size",
+		Short: "Calculate's Juno's DB size.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dbPath, err := cmd.Flags().GetString(dbPathF)
+			if err != nil {
+				return err
+			}
+
+			if dbPath == "" {
+				return fmt.Errorf("--%v cannot be empty", dbPathF)
+			}
+
+			pebbleDB, err := pebble.New(dbPath)
+			if err != nil {
+				return err
+			}
+
+			var totalSize, stateSizeWithoutHistory, stateSizeWithHistory uint
+
+			_, err = fmt.Fprintln(cmd.OutOrStdout(), "Total number of DB buckets:", len(db.BucketValues()))
+			if err != nil {
+				return err
+			}
+
+			var bucketSize uint
+			for _, b := range db.BucketValues() {
+				bucketSize, err = pebble.CalculatePrefixSize(cmd.Context(), pebbleDB.(*pebble.DB), []byte{byte(b)})
+				if err != nil {
+					return err
+				}
+
+				_, err = fmt.Fprintln(cmd.OutOrStdout(), uint(b)+1, "Size of", b, "=", bucketSize)
+				if err != nil {
+					return err
+				}
+
+				totalSize += bucketSize
+
+				if utils.AnyOf(db.StateTrie, db.ContractStorage, db.Class, db.ContractNonce,
```

Comment:
looks like `b` should be first argument, below as well

Reply:
Yep fixed